router/pkg/authentication/jwks_token_decoder.go

```diff
@@ -0,0 +1,47 @@
+package authentication
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/MicahParks/keyfunc/v2"
```

Comment:
Did you consider upgrading to v3?

https://github.com/MicahParks/keyfunc?tab=readme-ov-file#additional-features

Reply:
no, I'll upgrade to v3.

Reply:
I gave it a shot, but it turned out to be more complicated than expected. Do you still want me to do the upgrade?

Reply:
I'm ok with v2 if it well tested.